docs/clipboard-pickling/explainer.md

```diff
@@ -115,11 +115,33 @@ Sites can read or write pickled versions of sanitized formats, by providing the
 
-On MacOS (and iOS), clipboard formats are named using the UTI reverse-DNS naming convention. Therefore, a MIME type `"custom/format"` will be transformed to `"com.web.custom.format"`. Note that the pickling prefix `"com.web"` precedes the transformed format name `"custom.format"`, and the slash is converted to a period.
+On Windows, there is room for around [16,000 registered window messages and clipboard formats](https://devblogs.microsoft.com/oldnewthing/20150319-00/?p=44433). Once those are exhausted, things will start behaving erratically because window classes use the same pool of atoms as clipboard formats, so nobody will be able to register window classes until the user logs off and back on. Linux has a limitation on the atom space as well so we are proposing a new approach for custom format namings and payload serialization/deserialization format which will be safer from a security prespective and also deterministic as to how many custom formats are guranteed to be available in the clipboard inserted via pickling APIs. Generation of clipboard formats dynamically risks exhaustion of the atom pool.
 
-On Windows, clipboard formats are often named using capital words separated by a space. Therefore, a MIME type `"custom/format"` will be transformed to `"Web Custom Format`”. Note that the prefix `"Web"` precedes the transformed format name.
+There will be a custom format map which contains a JSON payload that will have a mapping of custom format MIME type to web custom format. There can only be 100 custom format per user session and it will be registered when the web authors request for a custom format. Since this format is allocated in the global atom space on Windows, even if the sites register it multiple times, the format strings will only be allocated once and `RegisterClipboardFormat` system function call will just return the unique value corresponding to that format.
 
-On Linux, ChromeOS, and Android, MIME types are often used (though Linux and ChromeOS don’t have strong recommendations regarding clipboard format naming conventions). Therefore, a MIME type `"custom/format"` will be transformed to `"application/web;type="custom/format""`. Note that the pickling prefix `"application/web;type="` precedes the transformed format name, which has quotes around the format name.
+Native apps will need to read the web custom format map and parse its payload as an alternative to `EnumFormats` on Windows to fetch the mapping of the custom MIME type to web custom format. Using this the native app can then request for the web custom format corresponding to a particular MIME type of interest. This also helps in [Delay Rendering](https://docs.microsoft.com/en-us/windows/win32/dataxchg/clipboard-operations#delayed-rendering) of formats on Windows where the native app just provides an indication to the clipboard that a format is available and when it is requested by some other app, it renders the content on-demand.
```

Comment:
* Delay Rendering -> Delayed Rendering, per the link.
* Please add commas or split up sentences where they get long.

Reply:
Done.